### PR TITLE
chore: migrate dependencies to version catalogs

### PR DESCRIPTION
**Pull request by romreviewer**
## Migration to Version suggested by Google For Better Maintainability.
[Gradle version catalogs](https://developer.android.com/build/migrate-to-catalogs) enable you to add and maintain dependencies and plugins in a scalable way. Using Gradle version catalogs makes managing dependencies and plugins easier when you have multiple modules. Instead of hardcoding dependency names and versions in individual build files and updating each entry whenever you need to upgrade a dependency, you can create a central version catalog of dependencies that various modules can reference in a type-safe way with Android Studio assistance.

### CODE REVIEW

File: app/build.gradle.kts [hunk 1]
```diff
@@ -1,10 +1,9 @@
 plugins {
-    id("com.android.application")
-    id("org.jetbrains.kotlin.android")
-    id("com.google.devtools.ksp")
+    alias(libs.plugins.android.application)
+    alias(libs.plugins.kotlin.android)
+    alias(libs.plugins.devtools)
     id("kotlin-parcelize")
     kotlin("plugin.serialization") version "1.8.21"
-    id("com.mikepenz.aboutlibraries.plugin") version "10.8.0"
```
Comment (Axelen123) [inline app/build.gradle.kts:7]: Why was this removed?

File: gradle/libs.versions.toml [hunk 2]
```diff
@@ -0,0 +1,97 @@
+[versions]
+ktx = "1.10.1"
+viewmodel-lifecycle = "2.6.1"
+splash-screen = "1.0.1"
+compose-activity = "1.7.2"
+paging = "3.1.1"
+work-runtime = "2.8.1ō"
+compose-bom = "2023.06.01"
+accompanist = "0.30.1"
+srialization = "1.5.1"
```
Comment (Axelen123) [inline gradle/libs.versions.toml:10]:
```suggestion
serialization = "1.5.1"
```

File: gradle/libs.versions.toml [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+# Accompanist
+accompanist-drawablepainter = { group = "com.google.accompanist", name = "accompanist-drawablepainter", version.ref = "accompanist" }
+accompanist-webview = { group = "com.google.accompanist", name = "accompanist-webview", version.ref = "accompanist" }
+accompanist-permissions = { group = "com.google.accompanist", name = "accompanist-permissions", version.ref = "accompanist" }
+accompanist-systemuicontroller = { group = "com.google.accompanist", name = "accompanist-systemuicontroller", version.ref = "accompanist" }
+accompanist-placeholder = { group = "com.google.accompanist", name = "accompanist-placeholder-material", version.ref = "accompanist" }
+
+# Kotlinx
+kotlinx-serilization-json = { group = "org.jetbrains.kotlinx", name = "kotlinx-serialization-json", version.ref = "srialization" }
```
Comment (Axelen123) [inline gradle/libs.versions.toml:57]:
```suggestion
kotlinx-serialization-json = { group = "org.jetbrains.kotlinx", name = "kotlinx-serialization-json", version.ref = "serialization" }
```

File: app/build.gradle.kts [hunk 4]
```diff
@@ -62,79 +61,70 @@ kotlin {
...
-    implementation("io.coil-kt:coil-compose:2.4.0")
-    implementation("me.zhanghai.android.appiconloader:appiconloader-coil:1.5.0")
+    implementation(libs.coil.compose)
+    implementation(libs.coil.appiconloader)
 
     // KotlinX
-    val serializationVersion = "1.5.1"
-    implementation("org.jetbrains.kotlinx:kotlinx-serialization-json:$serializationVersion")
-    implementation("org.jetbrains.kotlinx:kotlinx-collections-immutable:0.3.5")
+    implementation(libs.kotlinx.serilization.json)
```
Comment (Axelen123) [inline app/build.gradle.kts:94]:
```suggestion
    implementation(libs.kotlinx.serialization.json)
```

**Review by Axelen123 — CHANGES_REQUESTED** [review]

File: app/build.gradle.kts (hunk 1 above)
Comment (romreviewer) [reply app/build.gradle.kts:7]: Hi @Axelen123,
It is not removed just migrated to root build.gradle as mentioned in the official guide of [AboutLibraries](https://github.com/mikepenz/AboutLibraries) as well it should be in root build.gradle.
```
// Root build.gradle
id 'com.mikepenz.aboutlibraries.plugin' version "${latestAboutLibsRelease}" apply false

// App build.gradle
id 'com.mikepenz.aboutlibraries.plugin'
```

File: gradle/libs.versions.toml (hunk 2 above)
Comment (romreviewer) [reply gradle/libs.versions.toml]: fixed in 12bb8eed795ba4c8c105a1800f8579280255a93d

File: gradle/libs.versions.toml (hunk 3 above)
Comment (romreviewer) [reply gradle/libs.versions.toml]: fixed in 12bb8eed795ba4c8c105a1800f8579280255a93d

File: app/build.gradle.kts (hunk 4 above)
Comment (romreviewer) [reply app/build.gradle.kts]: fixed in 12bb8eed795ba4c8c105a1800f8579280255a93d

**Review by Axelen123 — APPROVED** [review]
Thank you for the contribution
